modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/util/KBDropdownItemsProvider.java

```diff
@@ -76,216 +91,81 @@ public KBDropdownItemsProvider(
 	}
 
 	public List<DropdownItem> getKBArticleDropdownItems(KBArticle kbArticle) {
-		return getKBArticleDropdownItems(kbArticle, null);
+		return getKBArticleDropdownItems(kbArticle, null, true, true, true);
+	}
+
+	public List<DropdownItem> getKBArticleDropdownItems(
+		KBArticle kbArticle, boolean historyEnabled, boolean printEnabled,
+		boolean subscriptionEnabled) {
```

Comment:
Instead of passing a bunch of booleans from the jsp, can you create private methods `_isHistoryEnabled`, etc. that read the request attribute? Then you'd only need to pass a single boolean `enableConfiguredActions` or similar to inform the provider if it has to honor de configuration or ignore it.

Reply:
I would prefer to leave it as it is now, my reasons are:
* I prefer to maintain the low coupling by not using request parameters, at least not so specific parameters as "init.jsp-enableKBArticleHistory". Other general parameters like "classNameId" or "selectedItemId" seems to be more correct.
* Your proposition just saves 2 extra parameters. Three additional parameters don't look too much to me.

Reply:
But the whole point of asking the provider for the items is so that the jsp doesn't have to worry about most details about those dropdowns. We need that extra boolean because in some cases you need to ignore the configuration (something that I really dislike, and I'm more than willing to make that disappear as well), so that may be acceptable. But putting in the JSP the responsibility to decide what menu items to include puts things totally backwards; that decission belongs in the display context/dropdown provider.